xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-macro/macroEditor.js

```diff
@@ -723,7 +724,10 @@ define(
           load.call(this, input, macroEditor);
         }
       });
-      submitButton.on('click', function(event) {
+      modal.find('.modal-content').on('submit', function(event) {
```

Comment:
I wonder if it wouldn't make sense for the modal to explicitly expose a form (or `formElement`) attribute to easily get the form without relying on, e.g., the fact that the form has the `model-content` class.

Reply:
If the `form` parameter is set to false (default), this element would not be a form but just a regular container for content, so the `formElement` name would not be appropriate. 
We can propose to expose a `modalContent`element as an improvement in another ticket, but from what I can see on https://github.com/search?q=org%3Axwiki%20.modal-content&type=code I doubt this would be very useful.

Let me know what you think about this. I can handle creating a small improvement ticket if you think it's worth it :)